src/Stratis.Bitcoin/Consensus/ConsensusManagerBehavior.cs

```diff
@@ -199,6 +198,38 @@ protected async Task ProcessGetHeadersAsync(INetworkPeer peer, GetHeadersPayload
             }
         }
 
+        protected ChainedHeader FindStartingHeader(ChainedHeader fork, uint256 hashstop)
+        {
+            ChainedHeader tip = this.consensusManager.Tip;
```

Comment:
Not the tip but the highest\last header to include. Sometimes it will be tip, sometimes hashstop or other header in between.